WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -701,268 +681,44 @@ private void updatePostSettingsAndSaveButton() {
         }
 
         @Override
         protected Address doInBackground(Double... args) {
             // args will be the latitude, longitude to look up
-            latitude = args[0];
-            longitude = args[1];
-
-            return GeocoderUtils.getAddressFromCoords(getActivity(), latitude, longitude);
-        }
-
-        protected void onPostExecute(Address address) {
-            setLocationStatus(LocationStatus.FOUND);
-            if (address == null) {
-                // show lat/long when Geocoder fails (ugly, but better than not showing anything
-                // or showing an error since the location has been assigned to the post already)
-                updateLocationText(Double.toString(latitude) + ", " + Double.toString(longitude));
-            } else {
-                String locationName = GeocoderUtils.getLocationNameFromAddress(address);
-                updateLocationText(locationName);
-            }
-        }
-    }
-
-    private class GetCoordsTask extends AsyncTask<String, Void, Address> {
-        @Override
-        protected void onPreExecute() {
-            setLocationStatus(LocationStatus.SEARCHING);
-            showLocationView();
-        }
-
-        @Override
-        protected Address doInBackground(String... args) {
-            String locationName = args[0];
-
-            return GeocoderUtils.getAddressFromLocationName(getActivity(), locationName);
-        }
-
-        @Override
-        protected void onPostExecute(Address address) {
-            setLocationStatus(LocationStatus.FOUND);
-            showLocationView();
-
-            if (address != null) {
-                double[] coordinates = GeocoderUtils.getCoordsFromAddress(address);
-                setLocation(coordinates[0], coordinates[1]);
-
-                String locationName = GeocoderUtils.getLocationNameFromAddress(address);
-                updateLocationText(locationName);
-            } else {
-                showLocationNotAvailableError();
-                showLocationSearch();
+            double latitude = args[0];
+            double longitude = args[1];
+            try {
+                return GeocoderUtils.getAddressFromCoords(getActivity(), latitude, longitude);
+            } catch (IllegalArgumentException iae) {
+                return null;
             }
         }
-    }
 
-    private LocationHelper.LocationResult locationResult = new LocationHelper.LocationResult() {
-        @Override
-        public void gotLocation(final Location location) {
-            if (getActivity() == null)
+        protected void onPostExecute(@Nullable Address address) {
+            if (address == null) {
                 return;
-            // note that location will be null when requesting location fails
-            getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    setLocation(location);
+            }
+            if (address.getMaxAddressLineIndex() > 0) {
```

Comment:
Nitpick: Should we move this check to the above if like below:

```
if (address == null || address.getMaxAddressLineIndex() == 0) {
    return;
}
```

And maybe change the comment to be inside that `if` and say something like: `If the address is not found, we should do nothing and keep the "latitude, longitude" format`. What do you think?